### ISSUE TITLE
New Rule: spacing around default parameters

### ISSUE THREAD

**Issue by wbecker** [eslint/eslint#3757]
There are already a bunch of spacing rules, but if I'm not mistaken there doesn't seem to be one around using default parameters:

At the moment there is no way for me to indicate that I prefer the style:

```javascript
function (param = "somevalue") {}
```

over

```javascript
function (param="somevalue") {}
```

and definitely never want some inconsistent aberration like

```javascript
function (param ="somevalue") {}
```

This is stylistic.
It should be core, as it effects core ES6 functionality and is consistent with other spacing rules.
I might be able to do this myself, if it is deemed to be useful.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by wbecker** [comment]
Actually - This seems to be handled by the space-infix-ops rule
